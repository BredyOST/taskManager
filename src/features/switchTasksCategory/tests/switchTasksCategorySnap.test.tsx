import {render} from "@testing-library/react";
import SwitchTasksCategory from "../switchTasksCategory.tsx";
import ActiveCategoryContextComponent
    from "../../../shared/context/activeCategoryContext/activeCategoryContextComponent.tsx";
import TaskContextComponent from "../../../shared/context/tasksContext/taskContextComponent.tsx";

jest.mock('../../../app/axios/taskApi/taskApi.ts', () => ({
    deleteTask: jest.fn(),
}))

jest.mock('../../../shared/hooks/useTasks.ts', () => ({
    useTasks: () => ({
        refetch: jest.fn()
    })
}))

jest.mock('../../../shared/ui/tabs/myTab.tsx', () => () => (<div>MyTab</div>))


describe('SwitchTasksCategory', () => {

    test('matches snapshot', () => {
        const {asFragment} = render(
            <ActiveCategoryContextComponent>
                <TaskContextComponent>
                    <SwitchTasksCategory />
                </TaskContextComponent>
            </ActiveCategoryContextComponent>
        );
        expect(asFragment()).toMatchSnapshot()
    })

    test('renders correct task count based on active category', () => {
        // const button = screen.getByRole('button', { name: /clear completed/i });
        // fireEvent.click(button);
        // expect(deleteTask).toHaveBeenCalledTimes(1);

    })

})
