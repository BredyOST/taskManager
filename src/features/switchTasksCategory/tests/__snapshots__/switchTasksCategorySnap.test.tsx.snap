// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchTasksCategory matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiToggleButtonGroup-root MuiToggleButtonGroup-horizontal sc-blHHSb ckxUuc css-1qb8o-MuiToggleButtonGroup-root"
    role="group"
  >
    <div
      class="sc-gtLWhw cNtLyJ"
    >
      <div
        class="sc-egkSDF"
      >
        0
      </div>
      <div
        class="sc-egkSDF"
      >
        items
      </div>
    </div>
    <div>
      MyTab
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary sc-ivxoEo JjZhe css-1emzoey-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      id=":r0:"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiLoadingButton-label css-27yv9x-MuiLoadingButton-label"
      >
        clear completed
      </span>
    </button>
  </div>
</DocumentFragment>
`;
